examples/websocket_tls/sticky/server/server.go

```diff
@@ -27,9 +28,25 @@ func onWebsocket(w http.ResponseWriter, r *http.Request) {
 		panic(err)
 	}
 	wsConn := conn.(*websocket.Conn)
-	wsConn.OnMessage(func(c *websocket.Conn, messageType websocket.MessageType, data []byte) {
-		// echo
-		c.WriteMessage(messageType, data)
+	mtx := sync.Mutex{}
+	var writer *websocket.LargeMessageWriter
+	wsConn.OnDataFrame(func(c *websocket.Conn, messageType websocket.MessageType, fin bool, data []byte) {
```

Comment:
This example is just effective for an echo service, not useful enough for general functional services.
An exported websocket.Conn.WriteFrame func seems better than LargeMessageWriter:
```golang
func (c *Conn) WriteFrame(messageType MessageType, firstFrame bool, fin bool, data []byte) error {
        return c.writeMessage(messageType, firstFrame, fin, data)
}
```
with WriteFrame, the application layer can easily handle echo message like this:
```golang
var isFirst bool = true
var frames [][]byte
wsConn.OnDataFrame(func(c *websocket.Conn, messageType websocket.MessageType, fin bool, data []byte) {
	err := c.WriteFrame(messageType, isFirst, fin, data)
	if err != nil {
		c.Close()
		return
	}
	isFirst = false
	if fin {
		isFirst = true
	}
})
```
or

```golang
var frames [][]byte
wsConn.OnDataFrame(func(c *websocket.Conn, messageType websocket.MessageType, fin bool, data []byte) {
	frames = append(frames, data)
	if fin {
		var err error
		for i, v := range frames {
			err = c.WriteFrame(messageType, i == 0, i == len(frames)-1, data)
			if err != nil {
				c.Close()
				return
			}
		}
		frames = nil
	}
})
```